Backpack/src/main/java/net/skyscanner/backpack/dialog/internal/BpkDialogImpl.kt

```diff
@@ -89,5 +96,33 @@ internal interface BpkDialogImpl {
     override fun addActionButton(view: View) {
       buttonsRoot?.addView(view, LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)
     }
+
+    fun addActionButton(button: BpkDialog.Button) {
+      val buttonsRoot = buttonsRoot
+      if (buttonsRoot != null) {
+        when (buttonsRoot.childCount) {
+          0 -> addActionButton(
+            createButton(
+              button = button,
+              type = when (type) {
+                BpkDialog.Type.Danger -> BpkButton.Type.Destructive
+                else -> BpkButton.Type.Featured
+              },
+            )
+          )
+          1 -> addActionButton(createButton(BpkButton.Type.Secondary, button))
```

Comment:
For destructive dialogs the second button should be link (and we shouldn't allow a third button)